lua/entities/gmod_wire_expression2/core/entity.lua

```diff
@@ -125,6 +134,24 @@ e2function entity entity:owner()
 	return getOwner(self, this)
 end
 
+__e2setcost(20)
+
+e2function table entity:keyvalues()
+	local ret = {n={},ntypes={},s={},stypes={},size=0} -- default table
+	if not IsValid(this) then return ret end
+	local keyvalues = this:GetKeyValues()
+	local size = 0
+	for k,v in pairs( keyvalues ) do
+		size = size + 1
+		ret.s[k] = v
+		ret.stypes[k] = string.lower(type(v)[1]) -- i swear there's a more elegant solution to this but whatever.
```

Comment:
Isn't there an e2 function that converts a type to an e2 typename? I'll try to find that.

Reply:
Guess there isn't. That sucks

Reply:
`E2Lib.guess_type`

Does `GetKeyValues` only contain string keys?